coreclient/migrations/20250703133517_message_status.sql

```diff
@@ -0,0 +1,34 @@
+-- SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+--
+-- SPDX-License-Identifier: AGPL-3.0-or-later
+--
+-- Add mimi id to conversation messages and a table for storing message statuses per message and user.
+--
+-- The combination of message id, user id and status is unique.
+--
+CREATE TABLE IF NOT EXISTS conversation_message_status (
+    message_id BLOB NOT NULL,
+    sender_user_uuid BLOB NOT NULL,
+    sender_user_domain TEXT NOT NULL,
+    -- Bit set of message statuses in the form `Sum_status 2**status`
+    status_bitset INT NOT NULL,
+    created_at TEXT NOT NULL,
+    PRIMARY KEY (message_id, sender_user_domain, sender_user_uuid),
+    FOREIGN KEY (message_id) REFERENCES conversation_messages (message_id) ON DELETE CASCADE
```

Comment:
We should (additionally) do deletes ourselves so that we catch all statuses even if the message was edited multiple times and the message id changed.